runtime-sdk/modules/evm/src/raw_tx.rs

```diff
@@ -0,0 +1,274 @@
+                message.gas_limit,
+            )
+        }
+        ethereum::TransactionV2::EIP1559(eth_tx) => {
+            let sig = k256::ecdsa::recoverable::Signature::new(
+                &k256::ecdsa::Signature::from_scalars(
+                    eth_tx.r.to_fixed_bytes(),
+                    eth_tx.s.to_fixed_bytes(),
+                )
+                .with_context(|| "signature from_scalars")?,
+                k256::ecdsa::recoverable::Id::new(eth_tx.odd_y_parity.into())
+                    .with_context(|| "recoverable id new")?,
+            )
+            .with_context(|| "recoverable signature new")?;
+            let message = ethereum::EIP1559TransactionMessage::from(eth_tx);
+
+            // `ethereum` crate is broken: it uses a list of the wrong length when encoding an
+            // EIP-1559 transaction's signing message.
+            // TODO: replace this with rlp::encode(&message)
+            let mut message_rlp = rlp::RlpStream::new_list(9);
+            message_rlp.append(&message.chain_id);
+            message_rlp.append(&message.nonce);
+            message_rlp.append(&message.max_priority_fee_per_gas);
+            message_rlp.append(&message.max_fee_per_gas);
+            message_rlp.append(&message.gas_limit);
+            message_rlp.append(&message.action);
+            message_rlp.append(&message.value);
+            message_rlp.append(&message.input);
+            message_rlp.append_list(&message.access_list);
+            let message_buf = message_rlp.out();
+
+            // `ethereum` crate is broken: it excludes the transaction type byte when hashing an
+            // EIP-1559 transaction.
+            // TODO: replace this with message.hash()
+            let mut sig_digest = sha3::Keccak256::new();
+            sig_digest.update(&[2]);
+            sig_digest.update(&message_buf);
+
+            // Base fee is zero. Allocate only priority fee.
+            let resolved_gas_price =
+                std::cmp::min(message.max_fee_per_gas, message.max_priority_fee_per_gas);
+            (
+                sig,
+                sig_digest,
+                message.action,
+                message.value,
+                message.input,
+                message.nonce,
+                resolved_gas_price,
+                message.gas_limit,
+            )
+        }
+    };
+    let (method, body) = match eth_action {
+        ethereum::TransactionAction::Call(eth_address) => (
+            "evm.Call",
+            cbor::to_value(types::Call {
+                address: eth_address.into(),
+                value: eth_value.into(),
+                data: eth_input,
+            }),
+        ),
+        ethereum::TransactionAction::Create => (
+            "evm.Create",
+            cbor::to_value(types::Create {
+                value: eth_value.into(),
+                init_code: eth_input,
+            }),
+        ),
+    };
+    let key = sig
+        .recover_verify_key_from_digest(sig_digest)
```

Comment:
This doesn't actually verify the signature so signature verification needs to be performed separately.

Reply:
I've asked about this, and I've read some vaguely related things to the contrary. can you send a citation for this?

Reply:
Looking at the function source I don't see a verification step:

https://docs.rs/k256/0.9.6/src/k256/ecdsa/recoverable.rs.html#190

Reply:
why does that have a question mark on it >_>

Reply:
mathematically though, does the recover operation sometimes give public keys for which the given signature is not valid?

Reply:
Hm, so in case the signature doesn't verify it would return a bogus key, but it would still continue? Because looking at that function it doesn't seem like it would fail in that case? E.g. the only failure that I can see is if decompression of `r` fails. Can you check what happens given a bogus signature?

Reply:
ok yeah

Reply:
yup, recovers to a different address

Reply:
I had to add a check for high `s` values ):

Reply:
tests added